sanitize_test.go

```diff
@@ -72,8 +99,49 @@ func Test_Sanitize_Options(t *testing.T) {
 			Input:  []string{time.RFC3339},
 			Output: time.RFC850,
 		},
+		OptionSanitizerFunc{
+			Name:      "capfirst",
+			Sanitizer: capFirst,
+		},
 	)
-	s.Sanitize(&d)
+
+	if err := s.Sanitize(&d); err != nil {
+		t.Fatalf("Sanitize() - got unexpected error %v", err)
+	}
+
+	if !reflect.DeepEqual(d, expected) {
+		t.Errorf("Sanitize() - got %+v but wanted %+v", d, expected)
+	}
+}
+
+func Test_Sanitize_InvalidCustomSanitizerType(t *testing.T) {
+	type Dog struct {
+		Name    string `san:"max=5,trim,lower"`
+		Adopted bool   `san:"capfirst"`
+	}
+
+	d := Dog{
+		Name:    "Borky Borkins",
+		Adopted: true,
+	}
+
+	expected := Dog{
+		Name:    "borky",
+		Adopted: true,
+	}
+
+	s, _ := New(
+		OptionSanitizerFunc{
+			Name:      "capfirst",
+			Sanitizer: capFirst,
+		},
+	)
+
+	if err := s.Sanitize(&d); err == nil {
+		t.Fatal("Sanitize() - did not receive expected error")
+	} else if !strings.Contains(err.Error(), "capfirst: invalid type") {
```

Comment:
Awesome, this is exactly why I asked for the test to be added. I wanted to see what the design was going to be when a tag was used for a custom sanitizer for an invalid type.